File: ios/Sources/Capacitor24hTimePlugin/Capacitor24hTimePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(Capacitor24hTimePlugin)
public class Capacitor24hTimePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "Capacitor24hTimePlugin"
    public let jsName = "Capacitor24hTime"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "is24h", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = Capacitor24hTime()



     @objc func is24h(_ call: CAPPluginCall) {
             call.resolve([
                "is24h": implementation.is24h()
            ])
           }
}

File: android/src/main/java/de/exitare/plugins/capacitor24htime/Capacitor24hTimePlugin.java
package de.exitare.plugins.capacitor24htime;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Capacitor24hTime")
public class Capacitor24hTimePlugin extends Plugin {

    private Capacitor24hTime implementation = new Capacitor24hTime();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

        @PluginMethod
            public void is24h(PluginCall call) {
                Boolean is24h = implementation.is24h(getContext());
                JSObject ret = new JSObject();
                ret.put("is24h", is24h);
                call.resolve(ret);
            }
}
